packages/std/src/coin.rs

```diff
@@ -19,6 +19,25 @@ impl Coin {
     }
 }
 
+impl FromStr for Coin {
+    type Err = StdError;
```

Comment:
Can we use a dedicated `CoinFromStrError` enum here and implement `From<CoinFromStrError> for StdError`? This way we can have more stuctured errors and avoid string allocations if they are handled low level.